scripts/ava-to-junit.mjs

```diff
@@ -0,0 +1,83 @@
+#! /usr/bin/env node
+import fs from 'fs';
+import process from 'node:process';
+
+const resultsBody = fs.readFileSync(process.argv[2] || 'results.txt', 'utf-8');
+const packageTestRegex = new RegExp(`> (.+?)@(.+?) test$`, 'gms');
+const resultRegex = new RegExp(
+  `^  ([✔✘-])(( \\[skip\\])?( \\[todo\\])?( \\[expected fail\\])?( \\[fail\\]:)?)(.+?)(\\(([\\d. smd]+)\\))?$`,
```

Comment:
Oof, this is really rough—and brittle against upstream changes like the rather recent https://github.com/avajs/ava/pull/3082 .

If the reason for not using TAP output is lack of timing information, I think we'd be better off adding that capability to the [local patch of ava](https://github.com/Agoric/agoric-sdk/blob/040ac7fe1c8856485ab51e6866c3f050d563c195/patches/ava%2B5.1.1.patch) (which we can even push for upstreaming).
```patch
diff --git a/node_modules/ava/lib/reporters/tap.js b/node_modules/ava/lib/reporters/tap.js
index b1989a4..eb30806 100644
--- a/node_modules/ava/lib/reporters/tap.js
+++ b/node_modules/ava/lib/reporters/tap.js
@@ -52,6 +52,7 @@ export default class TapReporter {
        this.extensions = options.extensions;
        this.stdStream = options.stdStream;
        this.reportStream = options.reportStream;
+       this.durationThreshold = options.durationThreshold || 100;

        this.crashCount = 0;
        this.filesWithMissingAvaImports = new Set();
@@ -96,7 +97,13 @@ export default class TapReporter {
    }

    writeTest(evt, flags) {
-       this.reportStream.write(supertap.test(this.prefixTitle(evt.testFile, evt.title), {
+       let title = this.prefixTitle(evt.testFile, evt.title);
+       if (evt.type === 'test-passed' && evt.duration > this.durationThreshold) {
+           // Add `# time=...` per TAP specification proposal
+           // https://github.com/TestAnything/Specification/issues/16
+           title += ` # time=${evt.duration}ms`;
+       }
+       this.reportStream.write(supertap.test(title, {
            comment: evt.logs,
            error: evt.err ? dumpError(evt.err) : null,
            index: ++this.i,
```

<details><summary>Sample output</summary>

```sh
$ npx ava --tap 2>/dev/null | tail
ok 67 - publish-kit › subscribeEach(publishKit subscriber)
ok 68 - publish-kit › subscribeEach(durablePublishKit subscriber)
ok 69 - publish-kit › durable publish kit upgrade trauma (full-vat integration) # time=20305ms

1..69
# tests 69
# pass 68
# skip 1
# fail 0
```

</details>

Reply:
```diff
diff --git a/node_modules/ava/lib/reporters/tap.js b/node_modules/ava/lib/reporters/tap.js
index b1989a4..0978a44 100644
--- a/node_modules/ava/lib/reporters/tap.js
+++ b/node_modules/ava/lib/reporters/tap.js
@@ -96,8 +96,8 @@ export default class TapReporter {
 	}
 
 	writeTest(evt, flags) {
-		this.reportStream.write(supertap.test(this.prefixTitle(evt.testFile, evt.title), {
-			comment: evt.logs,
+		this.reportStream.write(supertap.test(this.prefixTitle(evt.testFile, evt.title + ` dur=${evt.duration}`), {
+			comment: os.EOL + evt.logs,
 			error: evt.err ? dumpError(evt.err) : null,
 			index: ++this.i,
 			passed: flags.passed,
@@ -108,8 +108,8 @@ export default class TapReporter {
 
 	writeCrash(evt, title) {
 		this.crashCount++;
-		this.reportStream.write(supertap.test(title || evt.err.summary || evt.type, {
-			comment: evt.logs,
+		this.reportStream.write(supertap.test((title || evt.err.summary || evt.type) + ` dur=${evt.duration}`, {
+			comment: evt.duration + os.EOL + evt.logs,
 			error: evt.err ? dumpError(evt.err) : null,
 			index: ++this.i,
 			passed: false,
```